### PR TITLE
#305 fix

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
Change list:
- The #305 fix;
- update to Selenium 2.50.0
- changes of HasAppStrings interface. The old 'getAppStrings()' was returned with __Deprecated__ mark. Appium java client has wide user community. So let's care about users and let them prepare for serious API change. New methods were renamed to 'getAppStringMap'. @rgonalo Are you ok with this change?

### CODE REVIEW

**Comment by TikhomirovSergey** [comment]
@imurchie @bootstraponline 
Could you look at this PR before the merging?

**Comment by rgonalo** [comment]
:+1:
But, to be a non breaking change, we should return also as deprecated the old method `public String getAppStrings(String language)`

**Comment by TikhomirovSergey** [comment]
@rgonalo ok. Thanks for the reminding.

File: src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java [hunk 1]
```diff
@@ -36,14 +36,36 @@
...
     private static final String BIN_FOLDER = "bin";
+    private static final String BUILD_FOLDER = "build";
+    private static final String LIB_FOLDER = "lib";
+
     private static final String APPIUM_JS = "appium.js";
-    private static final String APPIUM_NODE_MASK =  File.separator +
-            APPIUM_FOLDER + File.separator + BIN_FOLDER + File.separator + APPIUM_JS;
+    private static final String MAIN_JS = "main.js";
+
+    private static final String ERROR_TEXT_WHEN_DEFAULT_APPIUM_NODE_IS_NOT_FOUND = "There is no installed nodes! Please " +
```
Comment (bootstraponline) [inline src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java]: `ERROR_TEXT_WHEN_DEFAULT_APPIUM_NODE_IS_NOT_FOUND` is too long. How about:

`ERROR_NODE_NOT_FOUND`

File: src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java [hunk 2]
```diff
@@ -36,14 +36,36 @@
 
 public final class AppiumServiceBuilder extends DriverService.Builder<AppiumDriverLocalService, AppiumServiceBuilder> {
 
+    /**
+     * The property of environmental variable used to define
```
Comment (bootstraponline) [inline src/main/java/io/appium/java_client/service/local/AppiumServiceBuilder.java]: `The property of` can be deleted and then it'll read more easily.

**Comment by bootstraponline** [comment]
code looks good to me :+1:

**Comment by TikhomirovSergey** [comment]
@rgonalo @bootstraponline 
ok. 
I'll fix it and merge it soon.

**Pull request merged**